app/resources/admin.html.template

```diff
@@ -138,559 +167,616 @@
     var title_col = row.insertCell(cur_col++);
     var title = document.createElement('input');
     title.type = 'text';
-    title.id = 'language_title_' + iteration;
+    title.id = table_id + '_language_value_' + iteration;
     title_col.appendChild(title);
-
     return iteration;
   }
 
+  function add_language_span(table_id, language) {
+    var table = $(table_id + '_table');
+    iteration = table.rows.length;
+    var row = table.insertRow(iteration);
+    var cur_col = 0;
+  
+    var selected_language_col = row.insertCell(cur_col++);
+    var selected_language = document.createElement('span');
+    var selected_language_text = document.createTextNode(language);
+    selected_language.className = 'language_selected_' + iteration;
+    selected_language.appendChild(selected_language_text);
+    selected_language_col.style.verticalAlign = 'top';
+    selected_language_col.appendChild(selected_language);
+
+    var title_col = row.insertCell(cur_col++);
+    var title = document.createElement('textarea');
+    title.cols = 72;
+    title.id = table_id + '_language_value_' + iteration;
+    title_col.appendChild(title);
+    return iteration; 
+  }
+
+  function add_another_language() {
```

Comment:
Doesn't have to be in this changeset, but I think it's a good timing to move to object-oriented Javascript programming as we are trying to do relatively complex stuff. It will help readability and maintainability.

var custom_message_tables = [];

function init_custum_message_tables() {
  custom_message_tables.push(new CustomMessageWidget($('repo_title'));
  custom_message_tables.push(new CustomLanguageMessageWidget($('start_page_custom_htmls'));
  ...
}

function add_language_to_custom_message_tables(language_code, language_name) {
  for (var i = 0; i < custom_message_tables.length; i++) {
    custom_message_tables[i].add_language(language_code, language_name);
  }
}

function remove_language_from_custom_message_tables(language_code) {
  for (var i = 0; i < custom_message_tables.length; i++) {
    custom_message_tables[i].remove_language(language_code);
  }
}

function validate_settings() {
  for (var i = 0; i < custom_message_tables.length; i++) {
    var table = custom_message_tables[i];
    $(table.id() + "_value").value = JSON.stringify(table.compile_value());
  }
}